File: solutions/python/perfect-numbers/1/perfect_numbers.py
def classify(number):
    if number < 1:
        raise ValueError("Classification is only possible for positive integers.")
    suma = 0
    for i in range(1, number//2 + 1):
        if not number % i:
            suma += i
    if number < suma:
        return "abundant"
    elif number == suma:
        return "perfect"
    else:
        return "deficient"

File: solutions/c/perfect-numbers/1/perfect_numbers.c
#include "perfect_numbers.h"

int classify_number(int number) {
    int suma = 0;
    if (number <= 0) {
        return -1;
    }
    for (int i = 1; i < number/2 +1; i++) {
         if ((number % i) == 0) {
             suma+=i;
         }
             
    }
    if (number == suma)
        return 1;
    if (number < suma)
        return 2;
    else return 3;
}